Polio 550 drag and drop component

olethanh opened the pull request:
The exiting `FileComponent` allowed drag and dropping files, but it wasn't clear to the user.

## Self proof reading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [X] Did I add translations


## Changes

- Created `SimpleModal`, `ConfirmCancelButtons`, `ConfirmCancelModal`  to replace `ConfirmCancelDialogComponent` and be able to make other types of modals (by composing `SimpleModal`). The main point was to remove `renderTrigger` so that when we conditionally render the modal component, the API calls are not made unless the modal is rendered.
- Created a `makeFullModal` HOC to facilitate making modals with `ConfirmCancelModal`
- Created `CustomInput` based on `OrgUnitTreeviewPicker`
- Added dependency to [react-dropzone](https://react-dropzone.js.org)
- Used `CustomInput` and react-dropzone to make `FileUpload` component: users can drag and drop files into the component. OnDragOver, it will show a dropzone. Users can also click on the input to open the file selection window
- Refactor `CreateEditBudgetEvent` to use `ConfirmCancelModal`, `makFullModal` and `FileUpload`



## How to test

- Go to a campaign's budget
- Click add step
- click on the "Files" input to open file selection window
- then drag and drop another selection of files in the component

To test the refactor:
- Open the modal to create a new step
- Fill it in
- Block `/api/polio/budgetfiles` and `/api/polio/confirm_budget`
- Confirm
- The table should show an additional icon
- click on it
- You should be able to resend files (but you have to re-authorize the endpoints first for it to work)

## Print screen / video

https://user-images.githubusercontent.com/38907762/189665826-73b498b2-ef85-4fc5-beb5-53a4336fa600.mov

## Notes

`SimpleModal`, `ConfirmCancelButtons`, `ConfirmCancelModal`,`makeFullModal` and `FileUpload` should be moved to bluesquare-components. I suggest to create a ticket after this PR is merged

quang-le committed:
POLIO-550 install react-dropzone

quang-le committed:
POLIO-550: create FileUpload component

quang-le committed:
POLIO-550: format dropzone

quang-le committed:
POLIO-550: move files

quang-le committed:
POLIO-550 add modalWithDnDField (WIP)

quang-le committed:
POLIO-550: WIP test new modal

quang-le committed:
POLIO-550: add ModalWithButton HOC

quang-le committed:
POLIO-550: have whole modal as DnD zone (WIP)

quang-le committed:
POLIO-550: add dragzone

quang-le committed:
POLIO-550: refactor CreateEditBudgetEvent
- use new modal
- use self contained fileUpload
- delete test components
- update ModalWithButton with iconProps

quang-le committed:
POLIO-550: typo

quang-le committed:
POLIO-550: add translations

beygorghor commented on hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx:
```diff
@@ -0,0 +1,145 @@
+import { Paper, InputLabel, Box } from '@material-ui/core';
+
+const childrenStyle = theme => ({
+    // replicated from .MuiTypography-body1 (except font-family and letter spacing)
```
so why not using typography directly ?

quang-le replied:
Because I needed to add the padding manually if I recall correctly

beygorghor commented on hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx:
```diff
@@ -0,0 +1,145 @@
+        marginLeft: '14px',
+        cursor: 'default',
+        color: 'transparent',
+        paddingTop: 0.5,
```
not sure what will be the result of this 0.5 px ??

beygorghor commented on hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx:
```diff
@@ -0,0 +1,145 @@
+            <InputLabel
+                shrink={Boolean(children)}
+                required={required}
+                className={`${classnames(
```
❤️

beygorghor commented on hat/assets/js/apps/Iaso/components/DragAndDrop/CustomInput.tsx:
```diff
@@ -0,0 +1,145 @@
+            <Paper
+                variant="outlined"
+                elevation={0}
+                className={`${classes.paper} ${enabledStyle} ${errorStyle} ${additionalPaperStyle}`}
```
😠

beygorghor commented on hat/assets/js/apps/Iaso/components/DragAndDrop/FilesUpload.tsx:
```diff
@@ -0,0 +1,103 @@
+};
+
+const Icon = (
+    <IconButton
```
So this is not a button, maybe use only ToolTip and the icon

beygorghor commented on plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx:
```diff
@@ -70,7 +78,7 @@ export const CreateEditBudgetEvent: FunctionComponent<Props> = ({
                         onSuccess: () => {
                             finalize(result.id, {
                                 onSuccess: () => {
-                                    closeModal.closeDialog();
```
👍

beygorghor commented on plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx:
```diff
@@ -208,7 +210,11 @@ export const CreateEditBudgetEvent: FunctionComponent<Props> = ({
                 maxWidth="sm"
                 cancelMessage={MESSAGES.cancel}
                 confirmMessage={MESSAGES.send}
-                renderTrigger={renderTrigger}
```
🥳

beygorghor commented on plugins/polio/js/src/pages/Budget/CreateEditBudgetEvent/CreateEditBudgetEvent.tsx:
```diff
@@ -270,14 +276,21 @@ export const CreateEditBudgetEvent: FunctionComponent<Props> = ({
-                                value={values.files}
-                                errors={getErrors('files')}
-                                label={MESSAGES.filesUpload}
+                            {/* <FileInputComponent
```
we can remove this then

beygorghor reviewed (approved):
Nice job ! only small remarks

quang-le committed:
POLIO-550: code review

- replace IConButton with Tooltip
- use classnames for legible styling
- remove commented code

quang-le committed:
Merge branch 'main' into POLIO-550_drag_and-drop_component
